toolchain/check/declaration_name_stack.cpp

```diff
@@ -73,15 +73,15 @@ auto DeclarationNameStack::LookupOrAddName(NameContext name_context,
             << name_context.unresolved_name_id << " in "
             << name_context.target_scope_id;
       }
-      return SemIR::NodeId::Invalid;
+      return SemIR::InstId::Invalid;
   }
 }
 
 auto DeclarationNameStack::AddNameToLookup(NameContext name_context,
-                                           SemIR::NodeId target_id) -> void {
-  auto existing_node_id = LookupOrAddName(name_context, target_id);
-  if (existing_node_id.is_valid()) {
-    context_->DiagnoseDuplicateName(name_context.parse_node, existing_node_id);
+                                           SemIR::InstId target_id) -> void {
+  SemIR::InstId existing_inst_id = LookupOrAddName(name_context, target_id);
```

Comment:
Preserve `auto` for now:
```suggestion
  auto existing_inst_id = LookupOrAddName(name_context, target_id);
```